test/integrations/destinations/salesforce/processor/data.ts

```diff
@@ -1608,4 +1608,228 @@ export const data = [
       },
     },
   },
+  {
+    name: 'salesforce',
+    description: 'Test 13 : Retry happens when access_token is undefined in secret',
+    feature: 'processor',
+    module: 'destination',
+    version: 'v0',
+    input: {
+      request: {
+        body: [
+          {
+            destination: {
+              Config: {
+                initialAccessToken: '7fiy1FKcO9sohsxq1v6J88sg',
+                password: 'dummyPassword2',
+                userName: 'test.c97-qvpd@force.com.test',
+                sandbox: true,
+              },
+              DestinationDefinition: {
+                DisplayName: 'Salesforce Sandbox',
+                ID: '1T96GHZ0YZ1qQSLULHCoJkow9KC',
+                Name: 'SALESFORCE_OAUTH_SANDBOX',
+              },
+              Enabled: true,
+              ID: '1ut7LcVW1QC56y2EoTNo7ZwBWSY',
+              Name: 'Test SF',
+              Transformations: [],
+            },
+            metadata: {
+              jobId: 1,
+              secret: {},
+            },
+            message: {
+              anonymousId: '1e7673da-9473-49c6-97f7-da848ecafa76',
+              channel: 'web',
+              context: {
```

Comment:
you can skip context object if not playing a role in the test case - helps in readability of test case